tests/CMakeLists.txt

```diff
@@ -237,9 +239,20 @@ add_h2_decoder_test_set(h2_decoder_headers)
 add_h2_decoder_test_set(h2_decoder_headers_padded)
 add_h2_decoder_test_set(h2_decoder_headers_priority)
 add_h2_decoder_test_set(h2_decoder_headers_ignores_unknown_flags)
+add_h2_decoder_test_set(h2_decoder_headers_request)
+add_h2_decoder_test_set(h2_decoder_headers_trailer)
+add_h2_decoder_test_set(h2_decoder_headers_empty_trailer)
 add_h2_decoder_test_set(h2_decoder_err_headers_requires_stream_id)
 add_h2_decoder_test_set(h2_decoder_err_headers_payload_too_small_for_padding)
 add_h2_decoder_test_set(h2_decoder_err_headers_payload_too_small_for_priority)
+add_h2_decoder_test_set(h2_decoder_malformed_headers_blank_name)
+add_h2_decoder_test_set(h2_decoder_malformed_headers_illegal_name)
+add_h2_decoder_test_set(h2_decoder_malformed_headers_response_to_server)
+add_h2_decoder_test_set(h2_decoder_malformed_headers_request_to_client)
+add_h2_decoder_test_set(h2_decoder_malformed_headers_mixed_pseudoheaders)
+add_h2_decoder_test_set(h2_decoder_malformed_headers_late_pseudoheaders)
+add_h2_decoder_test_set(h2_decoder_malformed_headers_trailer_must_end_stream)
+#add_h2_decoder_test_set(h2_decoder_malformed_header_continues_hpack_parsing)
```

Comment:
I uncovered a horrible bug in our HPACK code.
I didn't want it to hold up this PR though.